### ISSUE TITLE
Dropdown field name weirdness

### ISSUE THREAD

**Issue by missmatsuko** [barrelstrength/craft-sprout-forms#251]
### Description
I updated the form templates, but kept getting errors with just the dropdown form field. I figured out that the select input's name wasn't getting prefixed with `fields` and wrapped with square braces `[]`.

```
{{ dump(name) }} {# HomeType #}
{{ name }} {# HomeType #}
{%- if name %} name="{{ name }}"{% endif %} {# name="fields[HomeType]" #}
{{ name ? "name=" ~ name }} {# name="HomeType" #}
```
I pasted this code in the same area, so I don't think `name` is getting set to something else. Or at least it's not clear how.

I have no idea where the `fields` prefix and square brackets are coming from, and why they're only being attached in the 3rd case. I only noticed this happening on the dropdown field so far.

### Steps to reproduce
1. Create a form with dropdown fields and the accessible template.
2. Paste the code above somewhere in `vendor/barrelstrength/sprout-forms/src/templates/_components/formtemplates/accessible/fields/dropdown/input.twig`.

### Additional info

- Craft version: 3.1.15
- PHP version:
- Database driver & version:
- Plugins & versions:

**Comment by missmatsuko** [comment]
I think it's actually most/all of the fields now. I just had modified the dropdown field the most, so I noticed it first.

**Comment by BenParizek** [comment]
@missmatsuko Did you override the fields.html file?

The line that is setting the `fields[]` array around the fields is the [following line](https://github.com/barrelstrength/craft-sprout-forms/blob/v3/src/templates/_components/formtemplates/accessible/field.twig#L14):

```
{%- namespace field.getNamespace() -%}
```

**Comment by missmatsuko** [comment]
@BenParizek Yes, but I can reproduce this weirdness by just copying the code from the original post into the otherwise unmodified accessible templates. 

Is `namespace` part of Twig, or provided by the plugin? I couldn't find much about it.

I also noticed this:
```
id="{{ fieldName }}" {# id="fields-HomeType" #}
data-whatever="{{ fieldName }}" {# data-whatever="HomeType" #}
```

It seems like there's additional parsing being done on the Twig templates? It's causing some unexpected results.

**Comment by BenParizek** [comment]
Hi @missmatsuko 

My apologies for the slow reply here. `namespace` is a custom Twig Tag that Craft extends Twig and creates in Craft core. These types of Twig Extensions get pretty deep into how the code language is processed and are a bit confusing until you spend some time with them:
https://twig.symfony.com/doc/2.x/advanced.html#tags
https://docs.craftcms.com/api/v3/craft-web-twig-tokenparsers-namespacetokenparser.html
https://docs.craftcms.com/api/v3/craft-web-twig-nodes-namespacenode.html

If your curious, there is a Straight Up Hangout where I talk about it here in more detail with Brandon and Marion here:
https://straightupcraft.com/events/code-review-extending-twig-with-a-new-tag-and-the-mn-twig-perversion-plugin

The behavior you are seeing is logic designed into that tag which specifically looks for the HTML input `id` and `name` tags to process and namespace with the `fields`slug. Without going into it too much, this is used to help identify which fields are custom when processing Elements and saving things to the database.

In the case of your final example:

id="{{ 'some string' }}" {# id="fields-some string" #}

It appears that you are hard coding a space into the id and the namespace tag wrapped around that particular ID is trying to process it and not processing the space correctly. I think the only way to address that is to understand how the namespace tag works and plan your naming conventions around IDs within the input fields accordingly.

I'm not seeing any bugs here so I'm going to close this out, but feel free to follow up if you have a use case where you feel something is broken or not being considered. This is just how the namespace tag works and anything you do within it will have to take that into account.

**Comment by BenParizek** [comment]
Hi @missmatsuko 

My apologies for the slow reply here. `namespace` is a custom Twig Tag that Craft extends Twig and creates in Craft core. These types of Twig Extensions get pretty deep into how the code language is processed and are a bit confusing until you spend some time with them:
https://twig.symfony.com/doc/2.x/advanced.html#tags
https://docs.craftcms.com/api/v3/craft-web-twig-tokenparsers-namespacetokenparser.html
https://docs.craftcms.com/api/v3/craft-web-twig-nodes-namespacenode.html

If your curious, there is a Straight Up Hangout where I talk about it here in more detail with Brandon and Marion here:
https://straightupcraft.com/events/code-review-extending-twig-with-a-new-tag-and-the-mn-twig-perversion-plugin

The behavior you are seeing is logic designed into that tag which specifically looks for the HTML input `id` and `name` tags to process and namespace with the `fields`slug. Without going into it too much, this is used to help identify which fields are custom when processing Elements and saving things to the database.

In the case of your final example:

id="{{ 'some string' }}" {# id="fields-some string" #}

It appears that you are hard coding a space into the id and the namespace tag wrapped around that particular ID is trying to process it and not processing the space correctly. I think the only way to address that is to understand how the namespace tag works and plan your naming conventions around IDs within the input fields accordingly.

I'm not seeing any bugs here so I'm going to close this out, but feel free to follow up if you have a use case where you feel something is broken or not being considered. This is just how the namespace tag works and anything you do within it will have to take that into account.

**Comment by BenParizek** [comment]
Hi @missmatsuko 

My apologies for the slow reply here. `namespace` is a custom Twig Tag that Craft extends Twig and creates in Craft core. These types of Twig Extensions get pretty deep into how the code language is processed and are a bit confusing until you spend some time with them:
https://twig.symfony.com/doc/2.x/advanced.html#tags
https://docs.craftcms.com/api/v3/craft-web-twig-tokenparsers-namespacetokenparser.html
https://docs.craftcms.com/api/v3/craft-web-twig-nodes-namespacenode.html

If your curious, there is a Straight Up Hangout where I talk about it here in more detail with Brandon and Marion here:
https://straightupcraft.com/events/code-review-extending-twig-with-a-new-tag-and-the-mn-twig-perversion-plugin

The behavior you are seeing is logic designed into that tag which specifically looks for the HTML input `id` and `name` tags to process and namespace with the `fields`slug. Without going into it too much, this is used to help identify which fields are custom when processing Elements and saving things to the database.

In the case of your final example:

id="{{ 'some string' }}" {# id="fields-some string" #}

It appears that you are hard coding a space into the id and the namespace tag wrapped around that particular ID is trying to process it and not processing the space correctly. I think the only way to address that is to understand how the namespace tag works and plan your naming conventions around IDs within the input fields accordingly.

I'm not seeing any bugs here so I'm going to close this out, but feel free to follow up if you have a use case where you feel something is broken or not being considered. This is just how the namespace tag works and anything you do within it will have to take that into account.

**Comment by missmatsuko** [comment]
Oh, I haven't come across that before. Well, I guess if Craft is using it for their form templates... Sure makes it harder to use Payload Forwarding and add different IDs for JavaScript and internal anchors, though.

**Comment by BenParizek** [comment]
@missmatsuko We'll be releasing a new Integrations API soon which may help resolve some of that. I'll be happy to have your feedback once we do.

It really depends on how you need to send your data but if you don't need to store the data in Craft you could probably just remove that `namespace` tag from the templates and not deal with that adjustment. If you need to save the data in Craft, the fields do have to be formatted in a specific way. With the new API, you will have a bit more flexibility on submitting to Craft and then mapping your fields in some other way when forwarding the data.
